starfyre/parser.py

```diff
@@ -86,16 +101,25 @@ def handle_starttag(self, tag, attrs):
             else:
                 props[attr[0]] = attr[1]
 
+        # if the tag is not found in the generic tags but found in custom components
         if tag not in self.generic_tags and tag in self.components:
+            print(f'Tag = {tag} is not a generic BUT it is a custom component')
             component = self.components[tag]
             tag = component.tag
             component.props = {**component.props, **props}
             component.state = {**component.state, **state}
-            component.event_listeners = {**component.event_listeners, **event_listeners}
+            component.event_listeners = {
+                **component.event_listeners, **event_listeners}
             self.stack.append((component, self.current_depth))
+            print(f'{tag} is inside a Custom Tag. Stack is now = {self.stack}\n\n')
+
 
             return
 
+        # if the tag is not found in the generic tags and custom components
+        if tag not in self.generic_tags and tag not in self.components:
+            raise UnknownTagError(f'Unknown tag: "{tag}". Please review line {self.lineno} in your "{self.class_name}" component in the pyml code.')
```

Comment:
The parser throws the error message here

Reply:
Nicee